tasks/node_maintainance/node_health_monitor.py

```diff
@@ -0,0 +1,692 @@
+            result["result"] = False
+            result["reason"] = f"Cannot upsert node {ipaddr} with host-pool-consistency checks to server pool"
+            logger.error(result["reason"])
+            return result
+        logger.info(f"Document for node {ipaddr} with host-pool-consistency checks upserted to server pool successfuly")
+    except Exception as e:
+        result["result"] = False
+        result["reason"] = f"Cannot upsert node {ipaddr} with host-pool-consistency checks to server pool : {e}"
+        logger.error(result["reason"])
+        return result
+
+    result["result"] = True
+    result["ip_in_host_pool"] = doc["tags"]["ip_in_host_pool"]
+    if result["ip_in_host_pool"]:
+        result["origin_host_pool"] = doc["tags"]["origin_host_pool"]
+        result["vm_name_host_pool"] = doc["tags"]["vm_name_host_pool"]
+        result["os_version_host_pool"] = doc["tags"]["os_version_host_pool"]
+    return result
+
+def fetch_docs(poolId=[]):
+    result = {}
+    try:
+        server_pool_helper = ServerPoolSDKHelper()
+        logger.info(f"Connection to Server Pool successful")
+    except Exception as e:
+        result["result"] = False
+        result["reason"] = f"Cannot connect to Server Pool using SDK : {e}"
+        logger.error(result["reason"])
+        return result
+    try:
+        if len(poolId) > 0:
+            query_result = server_pool_helper.fetch_nodes_by_poolId(poolId)
+        else:
+            query_result = server_pool_helper.fetch_all_nodes()
+    except Exception as e:
+        result["result"] = False
+        result["reason"] = f"Cannot fetch all docs from server-pool : {e}"
+        logger.error(result["reason"])
+        return result
+
+    docs = []
+    for row in query_result:
+        row["_default"]["doc_key"] = row["id"]
+        docs.append(row["_default"])
+
+    result["result"] = True
+    result["docs"] = docs
+    return result
+
+def monitor_health_node(doc, tasks):
+    result  = {}
+    for task in tasks:
+        logger.info(f'{str(task.__name__)} for {doc["ipaddr"]}')
+        result[str(task.__name__)] = task(doc)
+    return result
+
+def monitor_health_nodes_parallel(docs, tasks, max_workers=None):
+    if not max_workers:
+        num_cores = os.cpu_count()
+        max_workers = num_cores if num_cores is not None else 10
+
+    final_result = {}
+    with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
+        futures = {executor.submit(monitor_health_node, doc, tasks): doc for doc in docs}
+        for future in concurrent.futures.as_completed(futures):
+            doc = futures[future]
+            result = future.result()
+            final_result[doc["ipaddr"]] = result
+            logger.critical(f"Deleting helper for remote {doc['ipaddr']}")
+            RemoteConnectionObjectFactory.delete_helper(ipaddr=doc["ipaddr"])
+
+    return final_result
+
+def fetch_and_monitor_health(tasks, poolId):
+    all_task_dic = {
```

Comment:
This can be made to a constant out of the function and moved to top of the file

Reply:
Addressed in latest commit